src/App.js

```diff
@@ -6,6 +6,9 @@ import Recipe from './components/Recipe.js'
 import Error from './components/Error.js'
 import SearchBar from './components/SearchBar'
 
+
+const branch = (test, ComponentOnPass, ComponentOnFail) => props => test ? props.errorMessage.map(error => <Error message={error} />) :  props.recipeList.map(recipe => <Recipe myRecipe = {recipe} />)
+
```

Comment:
It looks like you've got the concepts of a HOC down! Nice work 😄 - this is a really complicated concept!

One thing you could do to make this more generic is pass your props directly to your pass and fail components instead of manipulating the props here, similar to this:

``` jsx
const branch = (test, ComponentOnPass, ComponentOnFail) => props => 
test 
? < ComponentOnPass data={props} />
: <ComponentOnFail data={props} />
```
Then you can use the data however you'd like within the component itself, which means you could pass _any_ components to your `branch` component and reuse your `branch` component multiple times.

One other thing to double check is that you're passing your parameters into your HOC in the order you expect them - when you call this on line 67 you're passing  1) your condition, 2) your success component (Recipe) and 3) your fail component (Error). Here, you're rendering your `Error` component as the success component (which I think _is_ is what you actually want to render, since you're checking for `hasError`). It could be worth renaming your `test` boolean something like `passCondition` or similar so it's clear that the first returned component is the one you want rendered if your condition is true. And if you want your `Recipe` component to show with the success of your condition, you could change your boolean to `!this.state.hasError`.

Excellent work with Higher Order Components!